src/client/datascience/jupyter/jupyterExecution.ts

```diff
@@ -134,10 +134,10 @@ export class JupyterExecutionBase implements IJupyterExecution {
         return Cancellation.race(async () => {
             let result: INotebookServer | undefined;
             let connection: IJupyterConnection | undefined;
-            let kernelConnectionMetadata: KernelConnectionMetadata | undefined;
+            let kernelConnectionMetadata = options?.kernelConnection;
```

Comment:
Previously this was empty, and we ended up searching for a kernel all over again in the case of non-raw for Native notebooks (that was fixed in another PR, this code path was not fixed - opening an existing nb).